### PR TITLE
model details capture for azure openai

### PR DESCRIPTION
**Pull request by Hansrajr**
## Proposed changes

This PR - capturing the model details for azure openai 

## Types of changes

What types of changes does your code introduce to this project?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](https://github.com/monocle2ai/monocle/blob/main/CONTRIBUTING.md) doc
- [ ] I have signed the CLA
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: src/monocle_apptrace/metamodel/maps/langchain_methods.json [hunk 1]
```diff
@@ -46,6 +46,22 @@
         "wrapper_method": "allm_wrapper",
         "output_processor": ["metamodel/maps/attributes/inference/langchain_entities.json"]
     },
+    {
```
Comment (oi-raanne) [inline src/monocle_apptrace/metamodel/maps/langchain_methods.json:49]: similar changes to be made for llama index as well ?

File: tests/langchain_sample.py [hunk 2]
```diff
@@ -53,90 +67,219 @@ def format_docs(docs):
...
+#     },
+#     "attributes": {
+#         "span.type": "inference",
+#         "entity.count": 2,
+#         "entity.1.type": "inference.azure_oai",
+#         "entity.1.provider_name": "okahu-openai-dev.openai.azure.com",
+#         "entity.1.deployment": "kshitiz-gpt",
+#         "entity.1.inference_endpoint": "https://okahu-openai-dev.openai.azure.com/",
+#         "entity.2.name": "gpt-3.5-turbo-0125",
+#         "entity.2.type": "model.llm.gpt-3.5-turbo-0125"
```
Comment (oi-raanne) [inline tests/langchain_sample.py:209]: assert on this property

File: src/monocle_apptrace/metamodel/maps/langchain_methods.json (hunk 1 above)
Comment (Hansrajr) [reply src/monocle_apptrace/metamodel/maps/langchain_methods.json:49]: for llama index azure openai and openai uses same base model , so not required.

File: tests/langchain_sample.py (hunk 2 above)
Comment (Hansrajr) [reply tests/langchain_sample.py:209]: done , added asserts

**Review by oi-raanne — APPROVED** [review]

**Review by kshitiz-okahu — APPROVED** [review]

**Pull request merged**